### ISSUE TITLE
Bug: Issues for code with inline comment that ends with question mark. 

### ISSUE THREAD

**Issue by jcoleman42** [ryantam626/jupyterlab_code_formatter#235]
**Describe the bug**
This is a followup to a closed issue, https://github.com/ryantam626/jupyterlab_code_formatter/issues/204

In version 1.4.10, the problem described in issue #204 is still happening but is a bit more restricted on what triggers the problem.  

Extra indenting seems to be restricted for when a line of code contains an inline comment that ends with a question mark.   

As an example, this code causes the extra indenting, 
```python
def f(x, y):
    z = x + y # a comment that ends with ?
    return z
``` 

And this code, which is valid python, gives an error message, pictured below.
```python
def f(x, y):
    z = x + y # a comment that ends with ?
```
![image](https://user-images.githubusercontent.com/52928199/152884277-8bbe823e-8f88-4524-a871-c0f8206e36ed.png)

For completeness, the following code does NOT trigger the extra indenting.
```python
def f(x, y):
    # a comment that ends with ?
    z = x + y
    return z
```

**Comment by ryantam626** [comment]
Sorry for the late reply, was extremely burnt out and just switched off.

Thanks for the bug report, I think I have a fix but I ran out of time today, will be fixed in the coming days.

**Comment by ryantam626** [comment]
Thanks again for flagging this, I have re-implemented how escaping certain patterns prior to formatting is done, and this should be now fixed in 1.4.11 :pray:
